services/ipqsservicesample/service/ipqsclient.go

```diff
@@ -0,0 +1,161 @@
+package main
+
+import (
+  "github.com/gin-gonic/gin"
+  "github.com/gin-contrib/cors"
+  "net/http"
+  "io"
+  "fmt"
+  "encoding/json"
+)
+
+type PathBindings struct {
+  Email string `uri:"email"`
+  Phone string `uri:"phone"`
+  URL string `uri:"url"`
+  IP string `uri:"ip"`
+}
+
+type APIKey struct {
+  Key string `json:"key"`
+}
+
+const apiBase string = "https://www.ipqualityscore.com/api/json"
+var apiKey string
+
+func executeRequest(url string) (response []byte, err error) {
+  resp, err := http.Get(url)
+  if err != nil {
+    return nil, err 
+  }
+  
+  defer resp.Body.Close()
+  
+  body, err := io.ReadAll(resp.Body)
+  if err != nil {
+    return nil, err
+  }
+  
+  return body, nil
```

Comment:
Let's make sure we explore error handling a bit more and update both our service and our UI to handle things cleanly when issues crop up.